types/cometd/index.d.ts

```diff
@@ -81,12 +82,57 @@ export interface Configuration {
     useWorkerScheduler?: boolean;
 }
 
-export interface Message {
+export type ConnectionType = 'long-polling' | 'callback-polling' | 'iframe' | 'flash';
+export type ReconnectAdvice = 'retry' | 'handshake' | 'none';
+
+export interface BaseMessage {
     successful: boolean;
-    data: any;
+    channel: string;
+    id?: string;
+    clientId?: string;
+    advice?: {
+        reconnect?: ReconnectAdvice;
+        timeout?: number;
+        interval?: number;
+        'multiple-clients'?: boolean;
+        hosts?: string[];
+    };
+    connectionType?: ConnectionType;
+    timestamp?: string;
+    data?: any;
+    error?: string;
+    ext?: any;
+    version?: string;
+    minimumVersion?: string;
+}
+
+export interface SuccessfulHandshakeMessage extends BaseMessage {
+    successful: true;
+    version: string;
+    supportedConnectionTypes: ConnectionType[];
+    clientId: string;
+    authSuccessful?: true;
+    reestablish: boolean;
+}
+
+export interface UnsuccessfulHandshakeMessage extends BaseMessage {
+    successful: false;
+    error: string;
+    supportedConnectionTypes?: ConnectionType[];
+    reestablish?: undefined;
```

Comment:
this is a convenience so consumers can access `handshakeMessage.reestablish` without using a typeguard